external/storm-kafka/src/jvm/org/apache/storm/kafka/PartitionManager.java

```diff
@@ -199,7 +199,12 @@ private void fill() {
         try {
             msgs = KafkaUtils.fetchMessages(_spoutConfig, _consumer, _partition, offset);
         } catch (TopicOffsetOutOfRangeException e) {
-            offset = KafkaUtils.getOffset(_consumer, _partition.topic, _partition.partition, kafka.api.OffsetRequest.EarliestTime());
+            long partitionLatestOffset = KafkaUtils.getOffset(_consumer, _partition.topic, _partition.partition, kafka.api.OffsetRequest.LatestTime());
+            if (partitionLatestOffset < offset) {
+                offset = partitionLatestOffset;
+            }else{
```

Comment:
nit: `}else{` --> `} else {`   (i.e., please be consistent with other code for the spacing style)